src/pdf/ShoppingList/ShoppingListPDF.tsx

```diff
@@ -0,0 +1,223 @@
+        </View>
+    );
+};
+
+const ItemToRow: React.FC<Item> = (item) => {
+    return (
+        <View style={[styles.flexRow, styles.tableRow]}>
+            <View style={styles.tableItemDescription}>
+                <Text style={styles.normalText}>{item.description}</Text>
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.normalText}>{item.quantity}</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.normalText}>{item.notes}</Text>
+            </View>
+            <View style={styles.tableDone}></View>
+        </View>
+    );
+};
+
+const DisplayItemsList: React.FC<{ itemsList: Item[] }> = ({ itemsList }) => {
+    return (
+        <View>
+            {itemsList.map((item, index) => (
+                <ItemToRow {...item} key={index} />
+            ))}
+        </View>
+    );
+};
+
+const formatCamelCaseKey = (objectKey: string): string => {
+    return objectKey.replaceAll(/([a-z])([A-Z])/g, "$1 $2");
+};
+
+const DisplayAsBlock: React.FC<BlockProps> = <Prop extends BlockProps>(data: Prop) => {
+    return (
+        <View style={[styles.flexColumn, styles.infoCell]}>
+            {Object.keys(data).map((propKey, index) => (
+                <OneLine key={index} header={formatCamelCaseKey(propKey)} value={data[propKey]} />
+            ))}
+        </View>
+    );
+};
+
+const FormatClientCell: React.FC<{ propKey: string; propValue: string }> = ({
+    propKey,
+    propValue,
+}) => {
+    return (
+        <View style={[styles.flexColumn, styles.infoCell]}>
+            <View style={styles.normalText}>
+                <OneLine header={formatCamelCaseKey(propKey)} value={propValue} />
+            </View>
+        </View>
+    );
+};
+
+const DisplayClientSummary: React.FC<ClientSummary> = (clientSummary) => {
+    return (
+        <>
+            <View style={[styles.flexRow, styles.infoBlock]}>
+                <FormatClientCell propKey="name" propValue={clientSummary.name} />
+                <FormatClientCell propKey="contact" propValue={clientSummary.contact} />
+            </View>
+            <View style={[styles.flexRow, styles.infoBlock]}>
+                <FormatClientCell propKey="address" propValue={clientSummary.address} />
+                <FormatClientCell
+                    propKey="extraInformation"
+                    propValue={clientSummary.extraInformation}
+                />
+            </View>
+        </>
+    );
+};
+
+const ShoppingListPDF: React.FC<ShoppingListPDFProps> = ({
+    postcode,
+    parcelInfo,
+    clientSummary,
+    householdSummary,
+    requirementSummary,
+    itemsList,
+    endNotes,
+}) => {
+    return (
+        <Document>
+            <Page size="A4">
+                <View style={styles.paper}>
+                    <View style={[styles.flexRow, styles.pdfHeader]}>
+                        <View style={[styles.flexColumn, { padding: "5" }]}>
```

Comment:
Can you make the numbers consistent? Either all numbers or string number with units of pt or px.